sys/vane/ames.hoon

```diff
@@ -1438,17 +1077,23 @@
         ++  wool                                        ::    wool:ho:um:am
           |=  [hen=duct cha=path val=*]                 ::  send a statement
           ^+  +>
-          =+  ^=  rol  ^-  rill
-              =+  rol=(~(get by ryl.bah) cha)
-              ?~(rol *rill u.rol)
-          =+  sex=sed.rol
+          =/  rol=rill  (fall (~(get by ryl.bah) cha) *rill)
+          =/  sex=@ud  sed.rol
           ::  ~&  [%tx [our her] cha sex]
+          ::  if we don't have a public key for :her,
+          ::  request keys and proceed (skin will be %open)
+          ::
+          ::    XX update state so we only ask once?
```

Comment:
Yes, a "waiting on keys" is nice to have. In the "my sponsor doesn't have keys for her either" case, we just get the keys *eventually*, right?

Reply:
We don't currently set that state, so we re-request on every packet. But we request on the same duct, so the subscription is never duplicated. At least that used to be the case. With the more recent ames boot changes, the duct is reset on every restart. So this should be fixed.